acceptance/tests/base/dsl/helpers/host_helpers_test.rb

```diff
@@ -0,0 +1,1377 @@
+
+        rsync_to hosts.first, local_filename, remote_tmpdir
+
+        remote_filename = File.join(remote_tmpdir, "testfile.txt")
+        assert_raises Beaker::Host::CommandFailure do
+          remote_contents = on(hosts.first, "cat #{remote_filename}").stdout
+          # assert_equal "contents\n", remote_contents
+        end
+      end
+    end
+
+  else
+
+    step "#rsync_to fails if the local file cannot be found" do
+      remote_tmpdir = tmpdir_on hosts.first
+      assert_raises IOError do
+        rsync_to hosts.first, "/non/existent/file.txt", remote_tmpdir
+      end
+    end
+
+    step "#rsync_to CURRENTLY does not fail, but does not copy the file if the remote path cannot be found" do
+      # NOTE: would expect this to fail with Beaker::Host::CommandFailure
+
+      Dir.mktmpdir do |local_dir|
+        local_filename = File.join(local_dir, "testfile.txt")
+        File.open(local_filename, "w") do |local_file|
+          local_file.puts "contents"
+        end
+
+        rsync_to hosts.first, local_filename, "/non/existent/testfile.txt"
+        assert_raises Beaker::Host::CommandFailure do
+          on(hosts.first, "cat /non/existent/testfile.txt").exit_code
+        end
+      end
+    end
+
+    step "#rsync_to creates the file on the remote system" do
+      Dir.mktmpdir do |local_dir|
+        local_filename = File.join(local_dir, "testfile.txt")
+        File.open(local_filename, "w") do |local_file|
+          local_file.puts "contents"
+        end
+        remote_tmpdir = tmpdir_on hosts.first
+
+        rsync_to hosts.first, local_filename, remote_tmpdir
+
+        remote_filename = File.join(remote_tmpdir, "testfile.txt")
+      end
+    end
+
+    step "#rsync_to creates the file on all remote systems when a host array is provided" do
+      Dir.mktmpdir do |local_dir|
+        local_filename = File.join(local_dir, "testfile.txt")
+        File.open(local_filename, "w") do |local_file|
+          local_file.puts "contents"
+        end
+
+        remote_tmpdir = tmpdir_on hosts.first
+        on hosts, "mkdir -p #{remote_tmpdir}"
+        remote_filename = File.join(remote_tmpdir, "testfile.txt")
+
+        rsync_to hosts, local_filename, remote_tmpdir
+
+        hosts.each do |host|
+          remote_contents = on(host, "cat #{remote_filename}").stdout
+          assert_equal "contents\n", remote_contents
+        end
+      end
+    end
+  end
+
+  if test_scp_error_on_close?
+    step "#create_remote_file fails when the remote path does not exist" do
+      assert_raises Beaker::Host::CommandFailure do
+        create_remote_file hosts.first, "/non/existent/testfile.txt", "contents\n"
+      end
+    end
+
+    step "#create_remote_file fails when the remote path does not exist, using scp" do
+      assert_raises Beaker::Host::CommandFailure do
+        create_remote_file hosts.first, "/non/existent/testfile.txt", "contents\n", { :protocol => 'scp' }
+      end
+    end
+  end
+
+
+  def el4_platform?(host)
```

Comment:
You know, these are pretty useful little addons - I think that we should consider rolling them into the host abstraction.